### PR TITLE
namespaces output plugin

### PR DESCRIPTION
**Pull request by xuebingjie1990**
output `namespaces` into .yaml file #298

### CODE REVIEW

File: looper/conductor.py [hunk 1]
```diff
@@ -62,6 +62,29 @@ def write_sample_yaml(namespaces):
...
+    :return dict: sample namespace dict
+    """
+    import yaml
+    # my_namespaces = namespaces["sample"].to_dict()
+    path = get_sample_yaml_path(namespaces)
+    my_namespaces = {}
+    for namespace, values in namespaces.items():
+        if namespace == "sample":
+            values.to_yaml(path)
+            my_namespaces.update({str(namespace):yaml.load(open(path))})
```
Comment (stolarczyk) [inline looper/conductor.py:79]: why did you decide to read the sample.yaml in and update? There can be a case when the sample yaml file is not there.

File: looper/conductor.py [hunk 2]
```diff
@@ -62,6 +62,29 @@ def write_sample_yaml(namespaces):
...
+        else:
+            my_namespaces.update({str(namespace):values.to_dict()})
+    
+    # print(my_namespaces)
+    with open(path,'w') as yamlfile:
+        yaml.dump(my_namespaces, yamlfile)
```
Comment (stolarczyk) [inline looper/conductor.py:85]: This would _overwrite_ the sample.yaml file. I don't think this was the goal here.

File: looper/conductor.py [hunk 3]
```diff
@@ -62,6 +62,29 @@ def write_sample_yaml(namespaces):
     sample.to_yaml(get_sample_yaml_path(namespaces), add_prj_ref=False)
     return {"sample": sample}
 
+def write_namespaces(namespaces):
```
Comment (stolarczyk) [inline looper/conductor.py:65]: A better name is "submission.yaml", as suggested in #298

File: looper/__init__.py [hunk 4]
```diff
@@ -18,7 +18,7 @@
 from .const import *
 
 from .conductor import write_sample_yaml_cwl, write_sample_yaml, \
-    write_sample_yaml_prj
+    write_sample_yaml_prj, write_namespaces
```
Comment (stolarczyk) [inline looper/__init__.py:21]: follow the plugin naming convention we've established. So that would be: "write_submission_yaml".

**Review by stolarczyk — CHANGES_REQUESTED** [review]
You need to change the plugin to accept an input that would indicate where to save the file, not overwrite sample.yaml

File: looper/conductor.py (hunk 1 above)
Comment (xuebingjie1990) [reply looper/conductor.py:79]: for the other namespaces, I can convert the object type to dict, then write to .yaml.  But for sample, I have't find a way to convert it to dict. so I first write the sample.yaml the load it as dict.

Comment (stolarczyk) [reply looper/conductor.py:79]: That's not the optimal way to do this. Let's abstract the conversion of `Sample` to `dict` away from the `to_yaml` method and use it.

Comment (xuebingjie1990) [reply looper/conductor.py:79]: > That's not the optimal way to do this. Let's abstract the conversion of `Sample` to `dict` away from the `to_yaml` method and use it.

do you want me to make the changes in `sample.py`? that's in `peppy` right?

Comment (stolarczyk) [reply looper/conductor.py:79]: yes, this is in peppy, I'll do this.

**Comment by stolarczyk** [comment]
line 79 in `conductor.py`:

```
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/looper/conductor.py:79: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
```

File: looper/conductor.py (hunk 1 above)
Comment (stolarczyk) [reply looper/conductor.py:79]: should be ready to use: https://github.com/pepkit/peppy/blob/e8ccfda73d8b5abab5b77ebae964d5ecb17ebcec/peppy/sample.py#L79-L126

**Comment by stolarczyk** [comment]
- [ ] we need to add a unit test for this feature before merging

File: looper/conductor.py (hunk 1 above)
Comment (nsheff) [reply looper/conductor.py]: So this looper release will require an updated release to peppy?

**Comment by nsheff** [comment]
Is this ready to be merged?

File: looper/conductor.py (hunk 1 above)
Comment (stolarczyk) [reply looper/conductor.py]: yes, as I noted in the main PR description

Comment (nsheff) [reply looper/conductor.py]: not for this PR...

**Comment by stolarczyk** [comment]
> Is this ready to be merged?

I believe @xuebingjie1990 has finished this. I'll work on a unit test for this.

**Review by stolarczyk — APPROVED** [review]

**Pull request merged**